src/components/Primary-Button/Button.styles.ts

```diff
@@ -0,0 +1,68 @@
+import { Button as MuiButton, CircularProgress } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import { SIZE_CONFIGS, VARIANT_COLORS, SHAPE_CONFIGS } from './Button.constants';
+
+import type { ButtonProps } from './Button.types';
+
+export const LoadingSpinner = styled(CircularProgress)(() => ({
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  color: 'inherit',
+}));
+
+export const ButtonContent = styled('span')<{ $loading: boolean }>(({ $loading }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  gap: '8px',
+  opacity: $loading ? 0 : 1,
+  transition: 'opacity 0.3s ease',
+}));
+
+export const StyledButton = styled(MuiButton)<ButtonProps>(({
```

Comment:
StyledButton is created from MuiButton and typed with ButtonProps, but there is no shouldForwardProp filter. Custom props like loading, shape, fullWidth (that are not valid DOM attributes) will be forwarded to the underlying DOM element, causing React console warnings and potential HTML attribute issues.

Suggestion:
- Use styled(MuiButton, { shouldForwardProp }) and filter out custom props (loading, shape, fullWidth, etc.). Example:
  const shouldForwardProp = (prop: string) => !['loading','shape','fullWidth'].includes(prop);
  export const StyledButton = styled(MuiButton, { shouldForwardProp })<ButtonProps>( ({ theme, ... }) => (/* styles */) );

This prevents invalid attributes from reaching the DOM.